flask_restutils/random_pk.py

```diff
@@ -17,49 +16,31 @@ def id_to_uuid(id_str):
     return uuid.UUID(bytes=uuid_bytes)
 
 
-class RandomPKComparator(ColumnProperty.Comparator):
-    def __eq__(self, other):
-        try:
-            uuid_str = str(id_to_uuid(other))
-        except ValueError:
-            return False
-        return self.__clause_element__() == uuid_str
+class RandomPKField(TypeDecorator):
+    impl = UUID(as_uuid=True)
+    python_type = str
 
+    def __init__(self, prefix, *args, **kwargs):
+        self.prefix = prefix
+        super(RandomPKField, self).__init__(*args, **kwargs)
 
-def RandomPK(id_field):
-    def getter(self):
-        value = getattr(self, id_field)
-        if value:
-            return uuid_to_id(uuid.UUID(value), self.get_id_prefix())
+    def process_bind_param(self, value, dialect):
+        return id_to_uuid(value) if value else None
```

Comment:
Can we catch exceptions here so we get an empty object back when trying to get an invalid object ID? This should do:

```
diff --git a/flask_restutils/random_pk.py b/flask_restutils/random_pk.py
index ef142ee..efd53af 100644
--- a/flask_restutils/random_pk.py
+++ b/flask_restutils/random_pk.py
@@ -25,7 +25,10 @@ class RandomPKField(TypeDecorator):
         super(RandomPKField, self).__init__(*args, **kwargs)
 
     def process_bind_param(self, value, dialect):
-        return id_to_uuid(value) if value else None
+        try:
+            return id_to_uuid(value) if value else None
+        except ValueError:
+            return None
 
     def process_result_value(self, value, dialect):
         return uuid_to_id(value, self.prefix) if value else None
```